openapi-cli/src/main/java/io/ballerina/openapi/generators/client/BallerinaUtilGenerator.java

```diff
@@ -190,12 +207,15 @@ public SyntaxTree generateUtilSyntaxTree() throws IOException {
         }
 
         List<ImportDeclarationNode> imports = new ArrayList<>();
-        if (!(functionNameList.size() == 0 ||
+        if (!(functionNameList.size() == 0 || functionNameList.contains(CREATE_MULTIPART_BODY_PARTS) ||
                 (functionNameList.size() == 1 && functionNameList.contains(GET_MAP_FOR_HEADERS)))) {
```

Comment:
`ballerina/url` module is only used in `getEncodedUri` function. Therefore we could simplify the if condition as below. 
```suggestion
        if (functionNameList.contains(GET_ENCODED_URI)) {
```

Reply:
Fixed in https://github.com/ballerina-platform/openapi-tools/commit/1eae7688b4b2792493641a66c733a3581b0f0ae4